cranelift/codegen/src/isa/aarch64/lower.rs

```diff
@@ -257,6 +252,44 @@ pub(crate) fn put_input_in_reg<C: LowerCtx<I = Inst>>(
     }
 }
 
+/// Lower an instruction input to a register
+///
+/// The given register will be extended appropriately, according to
+/// `narrow_mode` and the input's type. If extended, the value is
+/// always extended to 64 bits, for simplicity.
+pub(crate) fn put_input_in_reg<C: LowerCtx<I = Inst>>(
+    ctx: &mut C,
+    input: InsnInput,
+    narrow_mode: NarrowValueMode,
+) -> Reg {
+    let (in_regs, ty, is_const) = put_input_in_regs(ctx, input);
+    let reg = in_regs
+        .only_reg()
+        .expect("Multi-register value not expected");
+
+    extend_reg(ctx, ty, reg, is_const, narrow_mode)
+}
+
+/// Lower an instruction input to multiple regs
+pub(crate) fn put_input_in_regs<C: LowerCtx<I = Inst>>(
```

Comment:
Sorry, I forgot that this function has other callers than just `put_input_in_reg()` and the new signature is more inconvenient for them - this should just return `ValueRegs<Reg>`, so that line 1575 in `cranelift/codegen/src/isa/aarch64/lower_inst.rs`, for example, doesn't ignore anything, and then the function body should be moved to another internal, non-public function, which is also going to be called by `put_input_in_reg()` directly.

Reply:
This feels a lot less awkward than what we had before